crates/wasmparser/src/validator/operators.rs

```diff
@@ -1064,6 +1064,28 @@ where
         self.push_operand(sub_ty)
     }
 
+    /// Common helper for checking the types of global types accessed
+    /// atomically.
+    fn check_atomic_global_ty(&self, global_index: u32, supertype: Option<ValType>) -> Result<()> {
```

Comment:
Instead of passing in a `supertype` here it might be best to leave this without an argument for now. Then it could grow an argument when equality-related globals are required (e.g. compare-and-swap instructions). In the future this might be best represented as a boolean flag rather than a `supertype` so indicate that it's basically just indicating the top type is one of two choices: anyref or eqref.